### ISSUE TITLE
useless-suppression false positive with ungrouped-imports

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#2366]
### Steps to reproduce

Run pylint on the following code:

```python
# pylint: enable=useless-suppression
from pylint import run_pylint
import astroid
from pylint import run_pyreverse # pylint: disable=ungrouped-imports
```

### Current behavior

This message is issued:
```
4:0: I0021: Useless suppression of 'ungrouped-imports' (useless-suppression)
```
However, if the comment is removed from line 4, this message is issued:
```
4:0: C0412: Imports from package pylint are not grouped (ungrouped-imports)
```

### Expected behavior

Since there is an actual message being suppressed, I would expect to get no `useless-suppression` message.

### pylint --version output

```
pylint 2.0.1
astroid 2.0.1
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```

**Comment by PCManticore** [comment]
Thanks, I can reproduce this bug. While `useless-supression` is inoffensive in the sense that it is not consider towards any score or not considered as an error per se, this bug shows that there is some inconsistency somewhere.

**Comment by brycepg** [comment]
So this particular check as a quit early conditional before adding the message:
```python
if not self.linter.is_message_enabled('ungrouped-imports', import_node.fromlineno):
      continue
```
Removing the line fixes the issue. Do we want to remove these guards to allow `add_message` to go through or disable `useless-supression` for that check if locally disabled?

**Comment by bersbersbers** [comment]
This is still an issue with
```
pylint==2.5.0
astroid==2.4.0
```
on Python 3.8.2.

I believe my example code is functionally identical to the OP's:

```
"""False-positive useless-suppression."""
# pylint: enable=useless-suppression
from numpy import array
from pandas import DataFrame

if True and True:
    from numpy import zeros  # pylint:disable=ungrouped-imports

print(array, DataFrame, zeros)
```

This code gives
```
bug.py:7:0: I0021: Useless suppression of 'ungrouped-imports' (useless-suppression)
```

If you remove `# pylint:disable=ungrouped-imports`, you get
```
bug.py:7:4: C0412: Imports from package numpy are not grouped (ungrouped-imports)
```

**Comment by bersbersbers** [comment]
Please know that the example in https://github.com/PyCQA/pylint/issues/2366#issuecomment-621398783 *is* functionally different because according to https://github.com/PyCQA/pylint/issues/4212#issuecomment-819539155, the `ungrouped-imports` is actually unexpected there, so the `useless-suppression` is expected. I filed #4360.
